asyncua/server/internal_subscription.py

```
@@ -50,6 +50,13 @@ async def start(self):
         if self.data.RevisedPublishingInterval > 0.0:
             self._task = self.loop.create_task(self._subscription_loop())
 
+            @self._task.add_done_callback
+            def subscription_loop_done(task):
+                if not task.cancelled():
+                    e = task.exception()
+                    if e is not None:
+                        self.logger.error("Exception in subscription loop", exc_info=e)
+
```

It is the first time I see that kind of code: using callbacks just to display exceptions.
The original implementation was more reable. ButI know that sometimes exceptions in asyncio code are swallowed, I thought it was a python bug but maybe we do something wrong...

---

I had this in my worktree for some time already, but unfortunately can not be sure why I wrote it. I recall something. During application startup the start had been called but then something failed and the reference was lost and task was collected without event loop getting a chance to run the exception handler.

I usually make sure the exception from created tasks are handled at the task creator rather than in the task itself. But I am not sure now if it is supposed to allow collecting the exception in this case also. Do this may not be a solution but an incomplete draft.

---

I did not find nor remember why I was making this change. The function is marked low level, so no reason to start using it.